### PR TITLE
Homework with Generics

### PR DESCRIPTION
**Pull request by Cherkashyn**
Please, check my homework with generics

### CODE REVIEW

File: homework1/src/main/java/lesson5/Bank.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    private String address;
+
+    public Bank(long id, String name, String address) {
+        this.id = id;
+        this.name = name;
+        this.address = address;
+    }
+
+    public <T extends Assets> List<T> clientInfo(T asset){
+        List generalInfo = new ArrayList();
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson5/Bank.java]: What reason use there List? You also use list without generic.  In our task more better remove it

File: homework1/src/main/java/lesson5/Bank.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package lesson5;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Bank {
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson5/Bank.java]: Your class must be a generic

File: homework1/src/main/java/lesson5/Test.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package lesson5;
+
+public class Test {
+    public static void main(String[] args) {
+        Bank account1 = new Bank(0001, "Yuriy Cherkashyn", "Solomianska 18a");
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson5/Test.java]: Use Bank<T>

File: homework1/src/main/java/lesson5/Bank.java (hunk 1 above)
Comment (SergeyPasko) [reply homework1/src/main/java/lesson5/Bank.java]: Use List<Type>

Comment (SergeyPasko) [reply homework1/src/main/java/lesson5/Bank.java]: In this case toString better way

File: homework1/src/main/java/lesson5/Currency.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package lesson5;
+
+public class Currency implements Assets {
+    private String type = "cash";
```
Comment (SergeyPasko) [inline homework1/src/main/java/lesson5/Currency.java]: You have same logic in children of Assets - good reason use abstract class

Comment (Cherkashyn) [reply homework1/src/main/java/lesson5/Currency.java]: Agree with you  - a lot of same code in different classes

**Pull request merged**